CMakeLists.txt

```diff
@@ -0,0 +1,168 @@
+# CMakeLists.txt in root
+#
+# @directions:
+#   Build from root/build/. with commands:
+#   > cmake .. (add `-A x64` on Windows to compile in 64 bits, MSVC compiles in 32 bits by default)
+#   > cmake --build . --target install (add `--config Release` on Windows, MSVC compiles in Debug mode by default)
+#   > ctest (add `-C Release` on Windows + add -R {lib, py or num} to specify a specific set of tests)
+#
+# @bug
+#   *   Library test crashes on x86 due to readCSV(90, 91) realloc invalid next size
+#   *   Adapt PLATFORM_INSTALL_PREFIX to build x86 binary on x64 host: no built-in CMake variable for TARGET architecture
+#   *   CMake CMAKE_SYSTEM_PROCESSOR in tool chain file: error at build time
+#       see https://stackoverflow.com/questions/54151877/cmake-cmake-system-processor-in-tool-chain-file-error-at-build-time
+#
+# @todo
+#   Temporary test dir for parallel running
+
+cmake_policy(SET CMP0048 NEW)
+cmake_minimum_required(VERSION 3.13.0)  # support for add_compile_options and add_link_options
+project(funnel VERSION 0.1.0 LANGUAGES C)
+enable_testing()
+
+# Detect OS.
+if(CMAKE_SYSTEM_NAME MATCHES Windows)
+    set(WINDOWS 1)
+    message("Windows system detected.")
+elseif(CMAKE_SYSTEM_NAME MATCHES Linux)
+    set(LINUX 1)
+    message("Linux system detected.")
+elseif(CMAKE_SYSTEM_NAME MATCHES Darwin)
+    set(MACOSX 1)
+    message("Mac OS X system detected.")
+else()
+    message(WARNING "OS unknown!")
+endif()
+
+# Output build system directories.
+message("CMake source directory is: ${CMAKE_SOURCE_DIR}")
+message("CMake binary directory is: ${CMAKE_BINARY_DIR}")
+message("CMake test directory is: ${CMAKE_TEST_DIR}")
+
+# Set compiling and linking options (requires VERSION > 3.13.0).
+# Consider adding -std=c89 -pedantic to gcc options to improve portability.
+set(UNIX_COMPILE_FLAGS -Wall -Wfloat-equal -o3 -fPIC)
```

Comment:
@AntoineGautier  
UNIX_COMPILE_FLAGS, WIN_COMPILE_FLAGS
What does lines specify..why do we need to add them?
Can you explain why you  add them

Reply:
@devil-d3 
`-fPIC` is a recommended option for compiling & linking a shared library with `gcc`. It enables generation of position-independent code so that the library can readily be loaded at any address in memory.
The `-o3` (or `/O2`) flag is for speed optimization.
The warning flags are self-explanatory.